examples/summarize_timing.cpp

```diff
@@ -0,0 +1,131 @@
+// Example OTIO program that reads a timeline and then prints a summary
+// of the video clips found, including re-timing effects on each one.
+
+#include "util.h"
+
+#include <opentimelineio/clip.h>
+#include <opentimelineio/deserialization.h>
+#include <opentimelineio/effect.h>
+#include <opentimelineio/freezeFrame.h>
+#include <opentimelineio/gap.h>
+#include <opentimelineio/linearTimeWarp.h>
+#include <opentimelineio/timeline.h>
+#include <opentimelineio/transition.h>
+
+#include <iostream>
+
+namespace otio = opentimelineio::OPENTIMELINEIO_VERSION;
+namespace otime = opentime::OPENTIME_VERSION;
+
+void _summarize_effects(otio::SerializableObject::Retainer<otio::Item> const& item)
+{
+    for (auto effect : item.value->effects())
+    {
+        if (auto freezeFrame = dynamic_cast<otio::FreezeFrame*>(effect.value))
```

Comment:
per your notes elsewhere, this definitely begs a dynamic_retained_cast.

Reply:
Maybe something like:
```
template <class T, class U>
otio::SerializableObject::Retainer<T> dynamic_retainer_cast(otio::SerializableObject::Retainer<U> const& retainer)
{
    return dynamic_cast<T*>(retainer.value);
}
```

Reply:
yes, exactly that ~ I think I'm leaning to sending this over to OTIO as a PR. It's in some sense trivial, but it would be nice to do it the same everywhere

Reply:
It might also be nice to add the "->" operator to Retainer so things like this:
`timeline.value->tracks()`

Could be simplified to:
`timeline->tracks()`

Reply:
I like it

Reply:
@meshula Just double-checking, did you want me to open a PR for the dynamic_retainer_cast()? If so do you think the other variants like static_retainer_cast(), etc., are useful as well?

Reply:
Just so I understand, that would make this code:
```c++
if (auto freezeFrame = otio::dynamic_retainer_cast<otio::FreezeFrame*>(effect)) {
```

Reply:
Actually even a little bit simpler since the "*" isn't necessary:
```
if (auto freezeFrame = otio::dynamic_retainer_cast<otio::FreezeFrame>(effect)) {
```

Reply:
I just created a draft PR that add this:
https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/923